### PR TITLE
update SPM documentation for DML SQL Bind and baseline capture

### PR DESCRIPTION
**Pull request by eurekaka**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pulls?q=is%3Apr+is%3Aclosed+label%3Aepic%2Fsql-plan-management

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by qiancai** [comment]
@qw4990 would you please review the changes in this PR and provide your comments? Thanks!

File: sql-plan-management.md [hunk 1]
```diff
@@ -16,17 +16,61 @@ aliases: ['/docs-cn/dev/sql-plan-management/','/docs-cn/dev/reference/performanc
...
+-- 类型二的改写：去掉 `delete` 语句中的 `using` 关键字
+create global binding for
+    delete t1 from t1 join t2 on t1.a = t2.a
+using
+    delete t1 from t1 join t2 on t1.a = t2.a;
+```
+
+> **注意：**
+>
+> 在对带 `SELECT` 子查询的 `INSERT` / `REPLACE` 语句创建执行计划绑定时，需要将想要绑定的优化器 Hints 指定在 `SELECT` 子查询中，而不是 `INSERT` / `REPLACE` 关键字后，不然优化器 Hints 不会生效。
```
Comment (qw4990) [inline sql-plan-management.md:64]: 这个 case 要不来组正/反面例子? @eurekaka

**Review by qw4990 — APPROVED** [review]

**Review by qw4990 — COMMENTED** [review]
/LGTM

**Comment by TomShawn** [comment]
@yikeke Please review this PR, thanks!

File: sql-plan-management.md [hunk 2]
```diff
@@ -16,17 +16,77 @@ aliases: ['/docs-cn/dev/sql-plan-management/','/docs-cn/dev/reference/performanc
...
 
 ```sql
-CREATE [GLOBAL | SESSION] BINDING FOR SelectStmt USING SelectStmt;
+CREATE [GLOBAL | SESSION] BINDING FOR BindableStmt USING BindableStmt;
 ```
 
-该语句可以在 GLOBAL 或者 SESSION 作用域内为 SQL 绑定执行计划。在不指定作用域时，隐式作用域为 SESSION。被绑定的 SQL 会被参数化后存储到系统表中。在处理 SQL 查询时，只要参数化后的 SQL 和系统表中某个被绑定的 SQL 语句一致，并且系统变量 `tidb_use_plan_baselines` 的值为 `on`（其默认值为 `on`），即可使用相应的优化器 Hint。如果存在多个可匹配的执行计划，优化器会从中选择代价最小的一个进行绑定。
+该语句可以在 GLOBAL 或者 SESSION 作用域内为 SQL 绑定执行计划。目前支持的可创建执行计划绑定的 SQL 类型（BindableStmt）包括：`SELECT`，`DELETE`，`UPDATE` 和带有 `SELECT` 子查询的 `INSERT` / `REPLACE`。
+
+值得注意的是，这其中有两类特定的语法由于语法冲突不能创建执行计划绑定，例如：
```
Comment (yikeke) [inline sql-plan-management.md:24]:
```suggestion
其中，有两类特定的语法由于语法冲突不能创建执行计划绑定，例如：
```

File: sql-plan-management.md [hunk 3]
```diff
@@ -16,17 +16,77 @@ aliases: ['/docs-cn/dev/sql-plan-management/','/docs-cn/dev/reference/performanc
...
+    insert into t1 select /*+ use_index(@sel_1 t2, a) */ * from t2 where a > 1 and b = 1;
+
+-- Hint 不能生效的用法
+create global binding for
+    insert into t1 select * from t2 where a > 1 and b = 1
+using
+    insert /*+ use_index(@sel_1 t2, a) */ into t1 select * from t2 where a > 1 and b = 1;
+```
+
+如果在创建执行计划绑定时不指定作用域，隐式作用域 SESSION 会被使用。被绑定的 SQL 会被"标准化"后存储到系统表中。在处理 SQL 查询时，只要"标准化"后的 SQL 和系统表中某个被绑定的 SQL 语句一致，并且系统变量 `tidb_use_plan_baselines` 的值为 `on`（其默认值为 `on`），即可使用相应的优化器 Hint。如果存在多个可匹配的执行计划，优化器会从中选择代价最小的一个进行绑定。
```
Comment (yikeke) [inline sql-plan-management.md:82]:
```suggestion
如果在创建执行计划绑定时不指定作用域，隐式作用域 SESSION 会被使用。TiDB 优化器会将被绑定的 SQL 进行“标准化”处理，然后存储到系统表中。在处理 SQL 查询时，只要“标准化”后的 SQL 和系统表中某个被绑定的 SQL 语句一致，并且系统变量 `tidb_use_plan_baselines` 的值为 `on`（其默认值为 `on`），即可使用相应的优化器 Hint。如果存在多个可匹配的执行计划，优化器会从中选择代价最小的一个进行绑定。
```
建议改成主动，更清晰

File: sql-plan-management.md [hunk 4]
```diff
@@ -47,15 +107,15 @@ explain select * from t1, t2 where t1.id = t2.id;
...
 ```sql
 select * from t where a >    1
--- 参数化后：
-select * from t where a > ？
+-- 标准化后：
+select * from test . t where a > ?
 ```
 
-每个标准化的 SQL 只能同时有一个通过 `CREATE BINDING` 创建的绑定。对相同的标准化 SQL 创建多个绑定时，会保留最后一个创建的绑定，之前的所有绑定（创建的和演进出来的）都会被标记为已删除。但 session 绑定和 global 绑定仍然允许共存，不受这个逻辑影响。
+每个标准化的 SQL 只能同时有一个通过 `CREATE BINDING` 创建的绑定。对相同的标准化 SQL 创建多个绑定时，会保留最后一个创建的绑定，之前的所有绑定（创建的和演进出来的）都会被删除。但 session 绑定和 global 绑定仍然允许共存，不受这个逻辑影响。
```
Comment (yikeke) [inline sql-plan-management.md:118]: 这一段对“标准化”的解释建议移到第 82 行，紧跟着本文第一次出现“标准化”字样的地方

File: sql-plan-management.md [hunk 5]
```diff
@@ -131,7 +195,15 @@ SHOW [GLOBAL | SESSION] BINDINGS [ShowLikeOrWhere];
...
-开启自动绑定功能后，每隔 `bind-info-lease`（默认值为 `3s`）会遍历一次 Statement Summary 中的历史 SQL 语句，并为至少出现两次的 SQL 语句自动捕获绑定。
+开启自动绑定功能后，每隔 `bind-info-lease`（默认值为 `3s`）会遍历一次 Statement Summary 中的历史 SQL 语句，并为至少出现两次的 SQL 语句自动捕获绑定。绑定的执行计划为 Statement Summary 中记录执行这条语句时使用的执行计划。
+
+对于以下几种 SQL 语句，TiDB 不会自动捕获绑定：
+
+- EXPLAIN 和 EXPLAIN ANALYZE 语句；
+- TiDB 内部执行的 SQL 语句，比如统计信息自动加载使用的 SELECT 查询；
+- 存在手动创建的执行计划绑定的 SQL 语句；
+
+对于 `PREPARE` / `EXECUTE` 语句组，或通过二进制协议执行的查询，TiDB 会为真正的查询自动捕获绑定，而不是 `PREPARE` / `EXECUTE` 语句。
```
Comment (yikeke) [inline sql-plan-management.md:206]:
```suggestion
对于 `PREPARE` / `EXECUTE` 语句组，或通过二进制协议执行的查询，TiDB 会为真正的查询（而不是 `PREPARE` / `EXECUTE` 语句）自动捕获绑定。
```
这个意思？

File: sql-plan-management.md [hunk 6]
```diff
@@ -159,7 +231,7 @@ set global tidb_evolve_plan_baselines = on;
 
 > **注意：**
 >
-> 设置 global 系统变量在当前 session 并不会生效，只会在新创建的 session 生效。可以将 global 关键字替换为 session 使当前 session 开启演进功能。
+> 自动演进功能目前不是 GA(Generally Available) 状态，不推荐在生产系统打开该功能。
```
Comment (yikeke) [inline sql-plan-management.md:234]:
```suggestion
> 自动演进绑定功能目前不是 GA (Generally Available) 状态，不推荐在生产环境打开该功能。
```

File: sql-plan-management.md [hunk 7]
```diff
@@ -16,17 +16,77 @@ aliases: ['/docs-cn/dev/sql-plan-management/','/docs-cn/dev/reference/performanc
...
+-- Hint 不能生效的用法
+create global binding for
+    insert into t1 select * from t2 where a > 1 and b = 1
+using
+    insert /*+ use_index(@sel_1 t2, a) */ into t1 select * from t2 where a > 1 and b = 1;
+```
+
+如果在创建执行计划绑定时不指定作用域，隐式作用域 SESSION 会被使用。被绑定的 SQL 会被"标准化"后存储到系统表中。在处理 SQL 查询时，只要"标准化"后的 SQL 和系统表中某个被绑定的 SQL 语句一致，并且系统变量 `tidb_use_plan_baselines` 的值为 `on`（其默认值为 `on`），即可使用相应的优化器 Hint。如果存在多个可匹配的执行计划，优化器会从中选择代价最小的一个进行绑定。
 
 值得注意的是当一条 SQL 语句在 GLOBAL 和 SESSION 作用域内都有与之绑定的执行计划时，因为优化器在遇到 SESSION 绑定时会将 GLOBAL 绑定的执行计划丢弃，该语句在 SESSION 作用域内绑定的执行计划会屏蔽掉语句在 GLOBAL 作用域内绑定的执行计划。
```
Comment (yikeke) [inline sql-plan-management.md:84]:
```suggestion
值得注意的是，如果一条 SQL 语句在 GLOBAL 和 SESSION 作用域内都有与之绑定的执行计划，因为优化器在遇到 SESSION 绑定时会丢弃 GLOBAL 绑定的执行计划，该语句在 SESSION 作用域内绑定的执行计划会屏蔽掉语句在 GLOBAL 作用域内绑定的执行计划。
```

File: sql-plan-management.md [hunk 8]
```diff
@@ -16,17 +16,77 @@ aliases: ['/docs-cn/dev/sql-plan-management/','/docs-cn/dev/reference/performanc
...
+    select * from t t1 join t t2;
+
+-- 类型二：包含了 `using` 关键字的 `delete` 语句
+create global binding for
+    delete from t1 using t1 join t2 on t1.a = t2.a
+using
+    delete from t1 using t1 join t2 on t1.a = t2.a;
+```
+
+可以通过等价的 SQL 改写绕过这个语法冲突的问题，对于上述两个例子可以改写为：
```
Comment (yikeke) [inline sql-plan-management.md:40]:
```suggestion
可以通过等价的 SQL 改写绕过这个语法冲突的问题。例如，上述两个例子可以改写为：
```

**Review by yikeke — APPROVED** [review]
/lgtm

**Comment by qiancai** [comment]
@CharLotteiu, this PR is ready for EN PR creation. PTAL, thanks!

**Pull request merged**